cl/lib/email_backends.py

```diff
@@ -0,0 +1,235 @@
+            # Verify if recipient email address is banned.
+            banned_email = EmailFlag.objects.filter(
+                email_address=convert_list_to_str(to),
+                object_type=OBJECT_TYPES.BAN,
+            )
+
+            if banned_email.exists():
+                continue
+
+            # If the recipient's email address is not banned continue,
+            # otherwise the message is discarded.
+            # Retrieve plain normal body and html normal body
+            [plaintext_body, html_body] = get_email_body(
+                message,
+                plain="text/plain",
+                html="text/html",
+            )
+
+            # Compute attachments total size in bytes
+            attachment_size = 0
+            for attachment in email_message.attachments:
+                # An attachment is a tuple: (filename, content, mimetype)
+                # with attachment[1] we obtain the base64 file content
+                # so we can obtain the file size in bytes with len()
+                attachment_size = len(attachment[1]) + attachment_size
+
+            # Check if the message contains a small version
+            small_version = False
+            for part in message.walk():
+                if (
+                    part.get_content_type() == "text/plain_small"
+                    or part.get_content_type() == "text/html_small"
+                ):
+                    small_version = True
+                    continue
+
+            if small_version:
+                # If the message has a small version
+                small_only = EmailFlag.objects.filter(
+                    email_address=convert_list_to_str(to),
+                    object_type=OBJECT_TYPES.FLAG,
+                    flag=EmailFlag.SMALL_ONLY,
+                )
+                if (
+                    small_only.exists()
+                    or attachment_size > settings.MAX_ATTACHMENT_SIZE
+                ):
+                    # If email address is small_only_email flagged or
+                    # attachments exceed the MAX_ATTACHMENT_SIZE
+
+                    # Retrieve plain small body and html small body
+                    [plain_small_body, html_small_body] = get_email_body(
+                        message,
+                        plain="text/plain_small",
+                        html="text/html_small",
+                    )
+
+                    # Store small message in DB and obtain the unique
+                    # message_id to add in headers to identify the message
+                    stored_id = store_message(
+                        from_email=from_email,
+                        to=to,
+                        bcc=bcc,
+                        cc=cc,
+                        reply_to=reply_to,
+                        subject=subject,
+                        message=plain_small_body,
+                        html_message=html_small_body,
+                        headers=headers,
+                    )
```

Comment:
Would it make sense for this function and compose_message to take the `EmailMessage` as the one parameter, instead of doing it with so many parameters like this? Or, you could probably pass it like this:  `store_message(**EmailMessage.__dict__)`, but that's kind of ugly.

But it'd be nice not to have so many parameters on these functions.

Reply:
Of course, this is much better, I've changed this and compose_message functions to receive a `EmailMessage` or `EmailMultiAlternatives`, instead of many parameters.